### PR TITLE
Add docs on how to improve performance by mixing synced folder types

### PR DESCRIPTION
**Pull request by oxyc**
See https://github.com/geerlingguy/drupal-vm/issues/1242#issuecomment-289130181.

How's that? Or maybe under the synced folder section as it's more a feature than a performance improvement suggestion

### CODE REVIEW

**Comment by geerlingguy** [comment]
This placement seems appropriate to me.

**Comment by geerlingguy** [comment]
Going to do a little testing before merging, though (I trust you, but I want to see how much of an impact this could make for poor forlorn Windows non-10 devs.

**Comment by oxyc** [comment]
Don't merge just yet btw, I'll test this too and probably change the example to use drupal-project structure `drupal/web` rather than `drupal/public` so it's easier for people to understand.

**Comment by oxyc** [comment]
So it didn't work :) It seems that since we updated the examples to use `local_path: .` the excluded_path examples have been incorrect. You cant reference a parent directory in the path.

**Comment by oxyc** [comment]
These are the rsync executions

```sh
rsync --verbose --archive --delete -z --chmod=ugo=rwX --no-owner --no-group \
--rsync-path "sudo rsync" -e "ssh -p 2200 -o LogLevel=FATAL  -o ControlMaster=auto -o ControlPath=/var/folders/d8/8h53d56x645c5p32t6_44l_40000gn/T/ssh.160 -o ControlPersist=10m  -o IdentitiesOnly=yes -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i '/Users/cindy/.vagrant.d/insecure_private_key'" \
--exclude ".vagrant/" \
--exclude "private" \
--exclude ".git" \
--exclude "web/sites/default/files" \
--exclude "tmp" \
--exclude "config/drupal" \
/Users/cindy/Projects/Personal/drupal-vm/ vagrant@127.0.0.1:/var/www/drupal
```

```
find /var/www/drupal \
-path /var/www/drupal/private -prune \
-o -path /var/www/drupal/.git -prune \
-o -path /var/www/drupal/web/sites/default/files -prune \
-o -path /var/www/drupal/tmp -prune \
-o -path /var/www/drupal/config/drupal -prune \
-o '!' -type l -a '(' ! -user vagrant -or ! -group vagrant ')' \
-exec chown vagrant:vagrant '{}'
````

**Comment by geerlingguy** [comment]
I'm also investigating doing reverse shared folders (e.g. sync folder from guest to host, using sshfs or other options).

**Comment by yareckon** [comment]
Sharing guest to host is **great** for php performance on large codebases. We do it for some vagrant projects.  I would suggest samba / smb as a protocol since all hosts have passable clients for it.

The drawbacks are:
1. Not having access to the codebase when the VM is shut down (you can't check on something quick without waiting for it to boot)
1. All the compatibility, configuration and performance problems move to the IDE / text editor, which now has to not choke on the "remote" codebase, or use inappropriate features on it like SASS compilation or git management.

**Comment by geerlingguy** [comment]
@yareckon - Especially "performance problems move to the IDE / text editor" — global find-and-replace is an especially slow performer.

**Comment by geerlingguy** [comment]
@oxyc - I'm okay with merging this as-is, if the way it's currently written works.

Maybe we could also add a link to the article [Drupal VM on Windows - a fast container for BLT project development](https://www.jeffgeerling.com/blog/2017/drupal-vm-on-windows-fast-container-blt-project-development), which has some recommendations around reverse-mounting a Samba share, or using SSH sync like with PhpStorm?

File: docs/other/performance.md [hunk 1]
```diff
@@ -67,9 +67,10 @@ vagrant_synced_folders:
...
 
+  1. **Use PhpStorm or a reverse-mounted synced folder for working from inside the VM.** Read [Drupal VM on Windows - a fast container for BLT project development](https://www.jeffgeerling.com/blog/2017/drupal-vm-on-windows-fast-container-blt-project-development) for instructions and recommendations.
   1. **Install the `vagrant-winnfsd` plugin**. See the 'NFS' section later for more details and caveats.
-  2. **Use `smb` for the synced folder's type.**
-  2. **Use `rsync` for the synced folder's type.** This requires that you have `rsync` available on your Windows workstation, which you can get if you install a substitute CLI like [Cygwin](https://www.cygwin.com/) or [Cmder](http://cmder.net/).
+  1. **Use `smb` for the synced folder's type.**
+  1. **Use `rsync` for the synced folder's type.** This requires that you have `rsync` available on your Windows workstation, which you can get if you install a substitute CLI like [Cygwin](https://www.cygwin.com/) or [Cmder](http://cmder.net/).
```
Comment (oxyc) [inline docs/other/performance.md]: I recall a discussion somewhere where you preferred this way of numbering markdown lists.

Comment (geerlingguy) [inline docs/other/performance.md]: Yeah; it's easier to update them without renumbering stuff. Good catch!

**Pull request merged**